community/planning/biome_oauth_user_store.md

```diff
@@ -20,174 +22,284 @@ management systems. These systems can include public providers, such as Github
+
+* This store will keep track of the refresh token associated with an OAuth
+  session, which may be needed in the future to keep a user's session going.
+
+* This store will allow the Splinter REST API to provide a custom access token
+  that is associated with the OAuth access token. This is important because the
+  Splinter REST API should not directly expose the OAuth access token for
+  security reasons.
 
 ## Guide-level explanation
 [guide-level-explanation]: #guide-level-explanation
 
-To support this change, we will introduce a new store for saving the OAuth user
-information.  This OAuthUserStore will manage the linkage between a user ID and
-the information required about an OAuth user.
+This change will introduce a new store for saving the OAuth user session info.
+This `OAuthUserSessionStore` will manage the linkage between a user ID and the
+information about an OAuth user and their active sessions.
 
 The information required for the OAuth user is:
 
-* provider user identifier: this is the identifier for the user within the
-  OAuth2 provider's system.  For example, email address in Google or Github
-  account name.
-* access token: the currently valid OAuth2 access token
-* refresh token: a refresh token, if one is provided
-* provider id: this is an id of the provider used.
+* Subject identifier: the unique identifier for the user within the OAuth2
+  provider's system.
+* OAuth access token: an OAuth2 access token for validating the user's
+  authentication with the OAuth provider.
+* OAuth refresh token: an refresh token for getting a new access token from the
+  OAuth provider (this is optional as not all OAuth providers have expiring
+  access tokens).
+* Splinter access token: the access token created and issued by the Splinter
+  REST API for the authenticated user's session.
 
-This information will be associated with a user ID, which is required to exist
-in the UserStore.  Implementations backed by databases, such as PostgreSQL, may
-optimize how this requirement is checked.
+This information will be associated with a Biome user ID, which is a generated
+UUID.
 
-This store will be used when the authorization process is completed to provide a
-biome user ID back to the client.  Before the callback returns the information
-to the client, it should check for the existence of the OAuth user.  If one does
-not exist, a user ID and an OAuth user should be created in tandem.
+Two initial implementations will be provided: one with in-memory storage, and
+another backed by a database via the Diesel crate.
 
 ## Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
 
-The OAuthUserStore will be defined by the following trait, in the module
-`splinter::biome::oauth`:
+The `OAuthUserSessionStore` will be defined by the following trait, in the
+module `splinter::biome::oauth`:
 
 ```rust
-use crate::error::{InternalError, ConstraintViolation};
+use std::time::Instant;
```

Comment:
Instant is a little different than SystemTime.  Are you sure you want that over the other?

Reply:
Good catch; updated to `SystemTime`.